### PR TITLE
Automated Updates: support parents and dependency imports

### PR DESCRIPTION
**Pull request by cuixq**
This PR adds support for parents and dependency imports:
 - fetching and parsing upstream pom.xml files
 - then merging the data into Maven project
 - add `RequirementsFromOtherPOMs` to system-specific data to store requirements that we don't have access

Some of the Maven code should be provided by `deps.dev/util/maven` and `deps.dev/util/resolve` soon...

### CODE REVIEW

File: internal/resolution/manifest/maven.go [hunk 1]
```diff
@@ -136,10 +201,53 @@ func (m MavenManifestIO) Read(df lockfile.DepFile) (Manifest, error) {
 		EcosystemSpecific: MavenManifestSpecific{
 			Properties:                 properties,
 			RequirementsWithProperties: reqsWithProps,
+			RequirementsFromOtherPOMs:  otherRequirements,
 		},
 	}, nil
 }
 
+const MaxParent = 100
```
Comment (michaelkedar) [inline internal/resolution/manifest/maven.go:209]: is this a limit in Maven itself, or just for us?

**Review by michaelkedar — APPROVED** [review]
LGTM, but Maven makes my head hurt

File: internal/resolution/manifest/maven.go (hunk 1 above)
Comment (cuixq) [reply internal/resolution/manifest/maven.go:209]: this is just for us

**Review by oliverchang — APPROVED** [review]

File: internal/resolution/manifest/maven.go (hunk 1 above)
Comment (another-rex) [reply internal/resolution/manifest/maven.go:185]: Can you add a comment here clarifying that?

File: internal/resolution/manifest/maven.go [hunk 2]
```diff
@@ -173,57 +257,6 @@ func mavenOrigin(list ...string) string {
 	return result
 }
 
-func makeMavenDepType(dependency maven.Dependency, origin string) dep.Type {
```
Comment (another-rex) [inline internal/resolution/manifest/maven.go:176]: Just to check, this is removed because the logic has been moved to resolve right?

**Review by another-rex — APPROVED** [review]
LGTM, added some minor comments.

File: internal/resolution/manifest/maven.go (hunk 2 above)
Comment (cuixq) [reply internal/resolution/manifest/maven.go:176]: this is now in `deps.dev/util/resolve`, so yes if this is what you refer to by `resolve`

File: internal/resolution/manifest/maven.go (hunk 1 above)
Comment (cuixq) [reply internal/resolution/manifest/maven.go:187]: done

**Pull request merged**